lib/dialogue-agent/conversation.ts

```diff
@@ -571,24 +570,10 @@ export default class Conversation extends events.EventEmitter {
     }
 
     async saveLog() {
-        const dir = path.join(this._engine.platform.getWritableDir(), 'logs');
-        try {
-            fs.mkdirSync(dir);
-        } catch(e) {
-            if (e.code !== 'EEXIST')
-                throw e;
-        }
-        const logfile = path.join(dir, this.id + '.txt');
-        const serializer = new DialogueSerializer({ annotations: true });
-
-        const output = fs.createWriteStream(logfile);
-
-        serializer.pipe(output);
+        const db = this.engine.db.getLocalTable('conversation');
         for (const dialogue of this._log)
-            serializer.write({ id: this.id, turns: dialogue.turns.map((log) => log.turn) });
-        serializer.end();
-
-        await StreamUtils.waitFinish(output);
+            for (const turn of dialogue.turns)
+                await db.insertOne(this.id, turn.turn);
```

Comment:
this.id is the ID of the conversation, which is the same for each turn, and also in practice always the same for all dialogues. You need to make up a unique ID for each turn.

Also, I think we should write each turn to the database incrementally when the turn is added to the log if recording is enabled, rather than in one fell swoop when saveLog is called. This is more robust in case of errors. 

Also, now that the log file no longer exists, the code immediately after (which is used to show the log to the user for error reporting) is no longer correct. You'll need to add a method to read the log back from the database.